django/utils/translation/__init__.py

```diff
@@ -107,6 +107,12 @@ def lazy_number(func, resultclass, number=None, **kwargs):
         proxy = lazy(func, resultclass)(**kwargs)
     else:
         class NumberAwareString(resultclass):
+            def __bool__(self):
+                return bool(kwargs['singular'])
```

Comment:
Can you describe the reason for using 'singular' here and (as an example) not 'plural'? Code looks fine style-wise; I'm trying to understand the issue a bit more since translation isn't my expertise (If you want a review about that).